docs/proposals/deletions_object_storage.md

```diff
@@ -0,0 +1,66 @@
+
+*   **Accidental insertion of confidential data:** This is a scenario where a user accidentally inserts confidential data and wants to delete it immediately. In this case, the user expects their request to accelerate an immediate deletion of the series pertaining to the blocks concerned with that specific data to be deleted.
+*   **GDPR:** Masking data and eventual deletion is expected.
+*   **Deletions to sustain user requirements:** Let’s assume the user has some data which leads to some unexpected results or causes performance degradation (due to high cardinality) and the user wants to restore the previous data set-up for obtaining the desired results. In this scenario, the user would want to send a request to mask the data for the time being as there isn’t a high priority requirement to delete the data but eventually during the compaction the user can expect the data to be deleted by the compactor not leading to any major performance issues.
+*   **Helps achieving series based retention (e.g rule aggregation) [#903](https://github.com/thanos-io/thanos/issues/903).**
+
+## Goals
+
+*   Unblock users and allow series deletion in the object storage using tombstones.
+*   Allowing undeletes for a default time duration and extending it as per the user requirements.
+*   Dealing with already downsampled and to be downsampled blocks with tombstones.
+*   Performing all the above operations at admin level.
+
+## Proposed Approach
+
+*   We propose to implement deletions using the tombstones approach.
+*   To start off we use the Prometheus tombstone file format.
+*   A user is expected to enter the following details for performing deletions:
+    *   **label matchers**
+    *   **start timestamp**
+    *   **end timestamp** (start and end timestamps of the series data the user expects to be deleted)
+    *   **maximum waiting duration for performing deletions** where a default value is considered if explicitly not specified by the user. (only after this time passes the deletions are performed by the compactor in the next compaction cycle)
+*   The details are entered via a deletions API and they are processed by the compactor to create a tombstone file, if there’s a match for the details entered. Afterwards the tombstone file is uploaded to object storage making it accessible to other components.
+*   **Why Compactor**? : The compactor is one of the very few components (next to Sidecar, Receiver, Ruler) that has write capabilities to object storage. At the same time it is only ever running once per object storage, so that we have a single writer to the deletion tombstones.
+*   If the data with the requested details couldn’t be found in the storage an error message is returned back as a response to the request.
+*   Store Gateway masks the series on processing the tombstones from the object storage.
+*   **Perspectives to deal with Downsampling of blocks having tombstones:**
```

Comment:
` **Perspectives to deal with Downsampling of blocks having tombstones:**` Unfinished?

Reply:
Oh you mean `Compactions` maybe?

Reply:
My bad! :facepalm: